### ISSUE TITLE
Example: Levenberg-Marquadt Algorithm

### ISSUE THREAD

**Issue by max-vassili3v** [dlfivefifty/DualArrays.jl#15]
Per [Fitzgibbon's paper](https://arxiv.org/pdf/2212.10307): the optimisations of DualArrays.jl provide a way to speed up the Levenberg-Marquadt algorithm (LM algorithm), particularly useful in fitting small to medium neural networks. This is because the LM algorithm combines the speed of convergence of the Gauss-Newton method with the stability of gradient descent ([Yu and Wilamowski, 2010](https://www.eng.auburn.edu/~wilambm/pap/2011/K10149_C012.pdf)). An example of this on a neural network to showcase this could be good

**Comment by dlfivefifty** [comment]
Sounds like a great example. I’d suggest doing both a “classical” case (perhaps involving an ODE to get banded jacobians) as well as the NN case
